apprise/plugins/NotifyThreema.py

```diff
@@ -0,0 +1,368 @@
+            self.targets.append(
+                (ThreemaRecipientTypes.USER, self.user))
+
+        return
+
+    def send(self, body, title='', notify_type=NotifyType.INFO, **kwargs):
+        """
+        Perform Threema Notification
+        """
+
+        if len(self.targets) == 0:
+            # There were no services to notify
+            self.logger.warning('There were no Threema targets to notify')
+            return False
+
+        # error tracking (used for function return)
+        has_error = False
+
+        # Prepare our headers
+        headers = {
+            'User-Agent': self.app_id,
+            'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8',
+            'Accept': '*/*',
+        }
+
+        # Prepare our payload
+        _payload = {
+            'secret': self.secret,
+            'text': body.encode('utf-8'),
+        }
+
+        # Create a copy of the targets list
+        targets = list(self.targets)
+
+        while len(targets):
+            # Get our target to notify
+            key, target = targets.pop(0)
+
+            # Prepare a payload object
+            payload = _payload.copy()
+
+            # Set Target
+            payload[key] = target
+
+            # Some Debug Logging
+            self.logger.debug('Threema GET URL: {} (cert_verify={})'.format(
+                self.notify_url, self.verify_certificate))
+            self.logger.debug('Threema Payload: {}' .format(payload))
+
+            # Always call throttle before any remote server i/o is made
+            self.throttle()
+
+            try:
+                r = requests.post(
+                    self.notify_url,
+                    params=payload,
+                    headers=headers,
+                    verify=self.verify_certificate,
+                    timeout=self.request_timeout,
+                )
+
+                if r.status_code != requests.codes.ok:
+                    # We had a problem
+                    status_str = \
+                        NotifyThreema.http_response_code_lookup(
+                            r.status_code)
+
+                    self.logger.warning(
+                        'Failed to send Threema notification to {}: '
+                        '{}{}error={}'.format(
+                            target,
+                            status_str,
+                            ', ' if status_str else '',
+                            r.status_code))
+
+                    self.logger.debug(
+                        'Response Details:\r\n{}'.format(r.content))
+
+                    # Mark our failure
+                    has_error = True
+                    continue
+
+                # We wee successful
+                self.logger.info(
+                    'Sent Threema notification to %s' % target)
+
+            except requests.RequestException as e:
+                self.logger.warning(
+                    'A Connection error occurred sending Threema:%s '
```

Comment:
BTW seeing these errors as I am not sure how these are logged, do you think it makes sense to call it something like "Threema Gateway (simple mode)", as when later the E2E mode is implemented one could differentiate it from the simple mode?

Reply:
When/If E2E mode is implemented, these errors can be adjusted accordingly then, no worries here.